src/tlo/methods/healthburden.py

```diff
@@ -529,8 +543,61 @@ def log_df_line_by_line(key, description, df, force_cols=None) -> None:
             df=self.get_dalys(yld=yld_by_wealth, yll=yll_by_wealth),
             force_cols=self._causes_of_dalys,
         )
-
+        # 5) Log the prevalence of each disease
+        log_df_line_by_line(
+            key='prevalence_of_diseases',
+            description='Prevalence of each disease.',
+            df=self.prevalence_of_diseases,
+            force_cols=self.recognised_modules_names,
+        )
         self._years_written_to_log += [year]
+    def write_to_log_prevalence(self):
+        """Write to the log the prevalence of conditions .
+        N.B. This is called at the end of the simulation as well as at the end of each month, so we need to check that
+        the year is not being written to the log more than once."""
+        def log_df_line_by_line(key, description, df, force_cols=None) -> None:
```

Comment:
this is a copy-and-paste of the function of the same defined as above. It's better to refactor than duplicate: so I would propose refactoring the original to be a function on this module and then calling it from here, rather than redefining it.